pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go

```diff
@@ -288,15 +288,16 @@ func (sf *streamIngestionFrontier) Next() (
 			break
 		}
 
-		var err error
-		if _, err = sf.noteResolvedTimestamps(row[0]); err != nil {
+		if _, err := sf.noteResolvedTimestamps(row[0]); err != nil {
 			sf.MoveToDraining(err)
 			break
 		}
 
 		if err := sf.maybeUpdatePartitionProgress(); err != nil {
 			// Updating the partition progress isn't a fatal error.
 			log.Errorf(sf.Ctx(), "failed to update partition progress: %+v", err)
+			sf.MoveToDraining(err)
```

Comment:
Do we want to always surface progress update errors or do we need some logic to only surface errors if 1 in x attempts fails? Are there any higher level retries that will prevent the job from entering a fatal state?

Reply:
> Do we want to always surface progress update errors or do we need some logic to only surface errors if 1 in x attempts fails?

An error here would indicate that the pts record is unreachable, which seems problematic. Changefeeds also error out in [an analogous codepath](https://github.com/cockroachdb/cockroach/blob/e8b075b44e6bea78c01c596ce4a2a54f3c057422/pkg/ccl/changefeedccl/changefeed_processors.go#L1418). Though it might be worth retrying the pts update within `maybeUpdatePartitionProgress()`. I think cdc folks have been thinking about this. I'll ping them.

The job currently has [60 retries ](https://github.com/cockroachdb/cockroach/blob/df7e2231cc3110da122bfe313044722c4818eac4/pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go#L291)before it moves to a paused state.

Reply:
I am tempted to wait and see how CDC folks handle this [issue ](https://github.com/cockroachdb/cockroach/issues/98585), and keep the PR as is for 23.1 at least.

Reply:
If we don't want to add it to this PR, let's open an issue to add some retry policy here. I'm still of the opinion that overall this PR is moving us further in the correct direction.

Reply:
I say we merge as is :D

Reply:
sounds good to me

Reply:
added a follow up issue https://github.com/cockroachdb/cockroach/issues/103918